bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/Event.java

```diff
@@ -23,41 +23,46 @@
  * A single event.
  *
  * @author Michael Wodniok - Initial contribution
- * @author Andrew Fiddian-Green - Added support for event description
+ * @author Andrew Fiddian-Green - Added support for event
+ * @author Michael Wodniok - Moved CommandTag paring to separate method for doing it only on demand.
  */
 @NonNullByDefault
 public class Event implements Comparable<Event> {
-    public final List<CommandTag> commandTags = new ArrayList<CommandTag>();
     public final Instant end;
     public final Instant start;
+    @Nullable
     public final String title;
+    @Nullable
+    public final String description;
+    @Nullable
+    public final String location;
+    @Nullable
+    public final String comment;
+    @Nullable
+    public final String contact;
 
-    public Event(String title, Instant start, Instant end, String description) {
-        this.title = title;
-        this.start = start;
+    public Event(@Nullable String summary, Instant begin, Instant end, @Nullable String description,
+            @Nullable String location, @Nullable String comment, @Nullable String contact) {
+        this.title = summary;
+        this.start = begin;
         this.end = end;
+        this.description = description;
+        this.location = location;
+        this.comment = comment;
+        this.contact = contact;
 
-        if (description.isEmpty()) {
+        if (description == null || description.isEmpty()) {
             return;
         }
+    }
 
-        String[] lines = description.replace("<p>", "").replace("</p>", "\n").split("\n");
-        for (String line : lines) {
-            CommandTag tag = CommandTag.createCommandTag(line);
-            if (tag != null) {
-                commandTags.add(tag);
-            }
-        }
+    public Event(String title, Instant start, Instant end, String description) {
+        this(title, start, end, description, null, null, null);
     }
 
     @Override
     public String toString() {
-        String[] tagStrings = new String[this.commandTags.size()];
-        for (int i = 0; i < tagStrings.length; i++) {
-            tagStrings[i] = this.commandTags.get(i).toString();
-        }
-        return "Event(title: " + this.title + ", start: " + this.start.toString() + ", end: " + this.end.toString()
-                + ", commandTags: List(" + String.join(", ", tagStrings) + ")";
+        return "Event(title: " + this.title + ", start: " + this.start.toString() + ", end: " + this.end.toString();
```

Comment:
`toString()` should also include the added fields `description`, `location`, `comment` and `contact`.

Reply:
OK.